Feat/chatbot

Ansarinamrah opened the pull request.

Ansarinamrah committed:
compare module done

Ansarinamrah committed:
chatbot done

skhasan071 commented on lib/features/chatbot/presentation/chatbot_view.dart:
```diff
@@ -0,0 +1,350 @@
+                          ),
+                          const Spacer(),
+                          if (vm.resolvingCards)
+                            const SizedBox(
```
No need to give Sizedbox, SLoadingIndicatior 20 is enough

Ansarinamrah committed:
sized box removed

skhasan536 committed:
Refactor: Update view models and navigation in compare and detail features

This commit introduces several refactoring changes to the compare and detail features:

- **CompareViewModel:** Removed optional constructor parameter for `CompareDataSourceImpl` and initialized it directly.
- **CompareWithView:** Fixed a minor variable naming issue in the `_filteredSchools` method.
- **OverviewView:**
    - Removed unused local variables `_resolvedSchoolId` and `_resolved`.
    - Corrected the usage of `_vm` instead of `ovm` when accessing school name for Alumni route.
    - Replaced `CircularProgressIndicator` with `SLoadingIndicator` for better consistency.
    - Added `failure_ext.dart` import for error handling.
- **Router:**
    - Modified `ActivityView` and `AmenitiesView` route builders to no longer require `schoolId` and `schoolName` as extras. These views will now likely fetch data based on the `OverviewViewModel`'s state.
    - Removed ChangeNotifierProvider setup for `ActivitiesViewModel` and `AmenitiesViewModel` in the router, as this is likely handled within the respective views or their parent.

skhasan536 committed:
Refactor: Update view models and navigation in compare and detail features

This commit introduces several refactoring changes to the compare and detail features:

- **CompareViewModel:** Removed optional constructor parameter for `CompareDataSourceImpl` and initialized it directly.
- **CompareWithView:** Fixed a minor variable naming issue in the `_filteredSchools` method.
- **OverviewView:**
    - Removed unused local variables `_resolvedSchoolId` and `_resolved`.
    - Corrected the usage of `_vm` instead of `ovm` when accessing school name for Alumni route.
    - Replaced `CircularProgressIndicator` with `SLoadingIndicator` for better consistency.
    - Added `failure_ext.dart` import for error handling.
- **Router:**
    - Modified `ActivityView` and `AmenitiesView` route builders to no longer require `schoolId` and `schoolName` as extras. These views will now likely fetch data based on the `OverviewViewModel`'s state.
    - Removed ChangeNotifierProvider setup for `ActivitiesViewModel` and `AmenitiesViewModel` in the router, as this is likely handled within the respective views or their parent.

skhasan536 committed:
Refactor: Improve SchoolDetailView data fetching and state management

This commit refactors how `SchoolDetailView` fetches and manages school data.

Key changes:
- Modified `OverviewViewModel`:
    - Added a setter for the `school` property to trigger `notifyListeners()` when the school data changes.
    - Updated `getSchoolsById` to use the new `school` setter, ensuring UI updates when data is fetched.
    - Added debug print statements to trace data flow within `getSchoolsById`.
- Updated `SchoolDetailView`:
    - Data fetching for `getSchoolsById` is now initiated in `addPostFrameCallback` within `initState`. This ensures that the view model attempts to fetch data after the widget tree is built.
    - Error handling for `getSchoolsById` is now done directly in `SchoolDetailView` by checking the return value and showing an error if necessary.
- Updated `AppRouter`:
    - Removed the `ChangeNotifierProvider` wrapping `SchoolDetailView` for `OverviewViewModel`. The view model is now instantiated and managed within `SchoolDetailView` itself.
    - Added a debug print statement to log the `schoolId` during routing.
